### PR TITLE
React refactor phase 3

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] Updates babel to 
- [x] Build seems to work

@keybase/react-hackers

### CODE REVIEW

File: desktop/app/menu-helper.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // Can't tell which thread we're in so let's try both
-const electron = require('electron')
+import electron from 'electron'
```
Comment (chrisnojima) [inline desktop/app/menu-helper.js]: let's avoid mixing imports and requires from now on. If you absolutely need a require() you have to use require().default else you sometimes won't get the default exported var

File: desktop/webpack.config.base.js [hunk 2]
```diff
@@ -15,8 +15,9 @@ module.exports = {
       loader: 'babel',
       exclude: /(node_modules|\/dist\/)/,
       query: {
-        optional: ['runtime'],
-        stage: 2
+        cacheDirectory: true,
```
Comment (chrisnojima) [inline desktop/webpack.config.base.js]: babel6!

**Comment by MarcoPolo** [comment]
:+1:

File: desktop/webpack.config.base.js (hunk 2 above)
Comment (MarcoPolo) [inline desktop/webpack.config.base.js]: How do we clear the cache?

Comment (chrisnojima) [inline desktop/webpack.config.base.js]: I don't think we really need to. the filename the cache uses is based on a sha1 of the file contents and the options used so i think its pretty safe:
https://github.com/babel/babel-loader/blob/2fa087a8a48407d6e0b161de8254a92bfae260a8/lib/fs-cache.js#L130

Safe enough they're talking about having it be on by default:
https://github.com/babel/babel-loader/issues/199

If we run into problems we can turn it off

**Pull request merged**